e2e/nomostest/clean.go

```diff
@@ -78,15 +82,52 @@ func Clean(nt *NT, failOnError FailOnError) {
 	// The admission-webhook prevents deleting test resources. Hence we delete it before cleaning other resources.
 	removeAdmissionWebhook(nt, failOnError)
 
-	// RepoSync finalizers can block namespace deletion.
-	// Disable deletion-propagation and delete the finalizers.
+	// Delete the reconciler-manager to stop reconcilers from being updated/re-created
+	if err := deleteReconcilerManager(nt); err != nil {
+		if failOnError {
+			nt.T.Fatal(err)
+		} else {
+			nt.T.Error(err)
+		}
+	}
+	// Delete the reconcilers to stop syncing
+	if err := deleteReconcilersBySyncKind(nt, configsync.RepoSyncKind); err != nil {
+		if failOnError {
+			nt.T.Fatal(err)
+		} else {
+			nt.T.Error(err)
+		}
+	}
+	if err := deleteReconcilersBySyncKind(nt, configsync.RootSyncKind); err != nil {
+		if failOnError {
+			nt.T.Fatal(err)
+		} else {
+			nt.T.Error(err)
+		}
+	}
+	// Remove RSync finalizers to unblock namespace deletion.
 	if err := disableRepoSyncDeletionPropagation(nt); err != nil {
```

Comment:
Given the reconciler-manager and reconcilers are deleted before this, what does disabling the finalizers accomplish? It seems to me like this wouldn't do anything since the deployments are gone

Reply:
Finalizers blocking deletion is a built-in k8s capability.

Technically removing the annotation is unnessecary now that the reconcilers are gone, but removing the finalizer is still required to unblock deletion.